### PR TITLE
[Qwix] int8 config

### PR DESCRIPTION
**Pull request by khatwanimohit**
# Description

This PR adds an option to use Qwix for quantization and adds an int8 config similar to its AQT counterpart `_get_int8_quant_config`

AQT Config we are replacing in this PR:
```
def _get_int8_quant_config(config):
  drhs_bits = None
  drhs_accumulator_dtype = None
  drhs_local_aqt = None
  if config.quantization_local_shard_count != 0:
    drhs_bits = 8
    drhs_accumulator_dtype = jnp.int32
    drhs_local_aqt = aqt_config.LocalAqt(contraction_axis_shard_count=config.quantization_local_shard_count)
  return aqt_config.config_v3(
      fwd_bits=8,
      dlhs_bits=8,
      drhs_bits=drhs_bits,
      rng_type="jax.uniform",
      dlhs_local_aqt=None,
      drhs_local_aqt=drhs_local_aqt,
      fwd_accumulator_dtype=jnp.int32,
      dlhs_accumulator_dtype=jnp.int32,
      drhs_accumulator_dtype=drhs_accumulator_dtype,
  )
  ```

If the change fixes a bug or a Github issue, please include a link, e.g.,:
FIXES: b/123456
FIXES: #123456

*Notice 1:* Once all tests pass, the "pull ready" label will automatically be assigned.
This label is used for administrative purposes. Please do not add it manually.

*Notice 2:* For external contributions, our settings currently require an approval from a MaxText maintainer to trigger CI tests.

# Tests

Please describe how you tested this change, and include any instructions and/or
commands to reproduce.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [ ] I have performed a self-review of my code.
- [ ] I have necessary comments in my code, particularly in hard-to-understand areas.
- [ ] I have run end-to-end tests tests and provided workload links above if applicable.
- [ ] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: MaxText/layers/quantizations.py [hunk 1]
```diff
@@ -606,3 +610,38 @@ def remove_quantized_params(params, aqt_vars):
...
+  return rules
+
+
+def get_fp8_config():
+  rules = [
+      qwix.QuantizationRule(
+        module_path='.*',  # Apply to all modules
+        weight_qtype=jnp.float8_e5m2,
+        act_qtype=jnp.float8_e4m3,
+        act_static_scale=True,
```
Comment (BirdsOfAFthr) [inline MaxText/layers/quantizations.py]: This doesn't seem to be the recipe we checked in: https://github.com/AI-Hypercomputer/maxtext/blob/e969faabbb571285a51545530f34d8f0a9f237e9/MaxText/layers/quantizations.py#L412

Comment (khatwanimohit) [reply MaxText/layers/quantizations.py]: I have updated it with constant calibration method now

Comment (BirdsOfAFthr) [reply MaxText/layers/quantizations.py]: This still doesn't seem right.

We use different precisions in fwd and bwd. And how do you apply static bound for weights (in both fwd and bwd)? And per-tensor scaling?

Could you with Qwix team to translate that AQT FP8 recipe to Qwix?

Comment (liudangyi) [reply MaxText/layers/quantizations.py]: The `_get_aqt_fp8_default_config` translates to

```
qwix.QtRule(
    weight_qtype="float8_e4m3fn",
    act_qtype="float8_e4m3fn",
    bwd_qtype="float8_e5m2",
    use_original_residuals=True,  # dlhs/drhs_use_fwd_quant=False
    disable_channelwise_axes=True,  # per_tensor_scales

    # The following enables constant range.
    weight_calibration_method="fixed,-7,7",
    act_calibration_method="fixed,-7,7",
    bwd_calibration_method="fixed,-7,7",
)
```
